client/js/widgets/basicwidget.js

```diff
@@ -22,12 +22,22 @@ class BasicWidget extends Widget {
   applyDeltaToDOM(delta) {
     super.applyDeltaToDOM(delta);
     if(delta.activeFace !== undefined || delta.faces !== undefined) {
-      let face = this.faces()[this.get('activeFace')];
-      if(face !== undefined)
-        this.applyDelta(face);
+      if(this.previouslyActiveFace !== undefined) {
+        const undoDelta = {};
+        for(const property in this.previouslyActiveFace)
+          undoDelta[property] = this.state[property] !== undefined ? this.state[property] : null;
+        this.applyDeltaToDOM(undoDelta);
```

Comment:
I prefer the pattern that I used in dice to eliminate recursive calls to applyDeltaToDOM().

Reply:
Yeah, makes sense. Feel free to change it or I'll do it later.

Reply:
The recursion is a problem because the undoDelta conceivably could contain activeFace and/or faces properties causing an infinite loop. The alternative approach would be to simply ensure that the undoDelta does not have either of these properties.

Reply:
I decided to do the less invasive approach. Feel free to change it to your pattern but this took less time for me to implement.